src/logging.h

```diff
@@ -167,4 +168,50 @@ static inline void LogPrint(const BCLog::LogFlags& category, const Args&... args
     }
 }
 
+class ILogFilter
+{
+public:
+    virtual ~ILogFilter(){};
+    virtual bool filter() = 0;
+};
+
+// Implementation that logs at most every millis_ milliseconds
+class TimeThrottledFilter : public ILogFilter
+{
+private:
+    uint64_t millis_{0};
+    uint64_t last_time_millis_{0};
+
+public:
+    TimeThrottledFilter() = delete;
+    explicit TimeThrottledFilter(uint64_t millis) : millis_(millis)
+    {
+    }
+
+    virtual bool filter() override
```

Comment:
You can remove `virtual` `override` is the correct function identifier.

Reply:
Done